src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,786 @@
+
+        // Check that the operator set exists and the operator is registered to it
+        OperatorSet memory operatorSet = OperatorSet(msg.sender, params.operatorSetId);
+        bool isRegistered = _isRegistered(params.operator, operatorSet);
+        require(_operatorSets[operatorSet.avs].contains(operatorSet.id), InvalidOperatorSet());
+        require(isRegistered, NotMemberOfSet());
+
+        uint256 length = _operatorSetStrategies[operatorSet.key()].length();
+        IStrategy[] memory strategiesSlashed = new IStrategy[](length);
+        uint256[] memory wadSlashed = new uint256[](length);
+
+        // For each strategy in the operator set, slash any existing allocation
+        for (uint256 i = 0; i < length; i++) {
+            // 1. Get the operator's allocation info for the strategy and operator set
+            IStrategy strategy = IStrategy(_operatorSetStrategies[operatorSet.key()].at(i));
+            (StrategyInfo memory info, Allocation memory allocation) =
+                _getUpdatedAllocation(params.operator, operatorSet.key(), strategy);
+            strategiesSlashed[i] = strategy;
+
+            // 2. Skip if the operator does not have a slashable allocation
+            // NOTE: this "if" is equivalent to: `if (!_isAllocationSlashable)`, because the other
+            // conditions in this method are already true (isRegistered + operatorSetStrategies.contains)
+            if (allocation.currentMagnitude == 0) {
+                continue;
+            }
+
+            // 3. Calculate the amount of magnitude being slashed, and subtract from
+            // the operator's currently-allocated magnitude, as well as the strategy's
+            // max and encumbered magnitudes
+            uint64 slashedMagnitude = uint64(uint256(allocation.currentMagnitude).mulWadRoundUp(params.wadToSlash));
+            uint256 sharesWadSlashed = uint256(slashedMagnitude).divWad(info.maxMagnitude);
+            wadSlashed[i] = sharesWadSlashed;
+
+            allocation.currentMagnitude -= slashedMagnitude;
+            info.maxMagnitude -= slashedMagnitude;
+            info.encumberedMagnitude -= slashedMagnitude;
+
+            // 4. If there is a pending deallocation, reduce the pending deallocation proportionally.
+            // This ensures that when the deallocation is completed, less magnitude is freed.
+            if (allocation.pendingDiff < 0) {
+                uint64 slashedPending =
+                    uint64(uint256(uint128(-allocation.pendingDiff)).mulWadRoundUp(params.wadToSlash));
+                allocation.pendingDiff += int128(uint128(slashedPending));
+
+                emit AllocationUpdated(
+                    params.operator,
+                    operatorSet,
+                    strategy,
+                    _addInt128(allocation.currentMagnitude, allocation.pendingDiff),
+                    allocation.effectBlock
+                );
```

Comment:
I found this confusing because `_updateAllocationInfo` also emits the same type of event, but I see that this is specifically for the now-modified pending deallocation.
I think there may be semantic inconsistency with using this event in this way in addition to the other usage -- you may want to consider a separate event for this case, or at least clarify that events of both semantics will be processed smoothly?